src/foam/nanos/dig/DUGDAO.java

```diff
@@ -1,62 +0,0 @@
-/**
- * @license
- * Copyright 2018 The FOAM Authors. All Rights Reserved.
- * http://www.apache.org/licenses/LICENSE-2.0
- */
-
-package foam.nanos.dig;
-
-import foam.core.Detachable;
-import foam.core.FObject;
-import foam.core.X;
-import foam.dao.AbstractSink;
-import foam.dao.DAO;
-import foam.dao.ProxyDAO;
-import foam.nanos.auth.User;
-import foam.nanos.logger.Logger;
-import foam.util.Auth;
-
-/** Execute DUGs on startup or when first put(). Set DUG.owner to creator. **/
-public class DUGDAO
-  extends ProxyDAO
-{
-  public DUGDAO(X x) {
-    setX(x);
-  }
-
-  public DUGDAO(X x, DAO delegate) {
-    setX(x);
-    setDelegate(delegate);
-  }
-
-  public void setDelegate(DAO delegate) {
-    super.setDelegate(delegate);
-
-    delegate.select(new AbstractSink() {
-      @Override
-      public void put(Object obj, Detachable sub) {
-        executeDUG(getX(), (DUG) obj);
-      }
-    });
-  }
-
-  @Override
-  public FObject put_(X x, FObject obj) {
-    User user = (User) x.get("user");
-    DUG  dug  = (DUG) obj;
-
-    dug.setOwner(user.getId());
-    executeDUG(x, dug);
-
-    return super.put_(x, obj);
-  }
-
-  protected void executeDUG(X x, DUG dug) {
-    try {
-      dug.execute(Auth.sudo(x, dug.getOwner()));
```

Comment:
@jlhughes do we need to preserve the user context when dug is run.

Reply:
One of the old behaviors of DUG is that it switches to dug.owner context before getting the target dao from that context then add an HTTPSink listener.

For DUG rule, we don't need to get the target dao since the rule engine already does that for us and hands back the `obj` and the thing left to be done is putting it to an HTTPSink as seen in DugRuleAction.

Reply:
sudo in this case makes sense.  The object will be processed as per the rule owners privileges.